src/kOS/Screen/TermWindow.cs

```diff
@@ -364,13 +385,47 @@ void TerminalGui(int windowId)
                 ShowCharacterByAscii((char)1, screen.CursorColumnShow, screen.CursorRowShow, currentTextColor);
             }
             GUI.EndGroup();
+            
+            GUI.Label(new Rect(windowRect.width/2-40,windowRect.height-20,100,10),screen.ColumnCount+"x"+screen.RowCount);
 
+            CheckResizeDrag(); // Has to occur before DragWindow or else DragWindow will consume the event and prevent drags from being seen by the resize icon.
+            GUI.DragWindow();
         }
-
+        
+        protected void CheckResizeDrag()
+        {
+            if (Input.GetMouseButton(0)) // mouse button is down
+            {
+                if (resizeMouseDown) // and it's in the midst of a drag.
+                {
+                    Vector2 dragDelta = mousePosAbsolute - mouseButtonDownPosAbsolute;
+                    windowRect = new Rect(windowRect.xMin,
+                                          windowRect.yMin,
+                                          System.Math.Max(resizeOldSize.x + dragDelta.x, 200),
+                                          System.Math.Max(resizeOldSize.y + dragDelta.y, 200));
+                }
+            }
+            else // mouse button is up
+            {
+                if (resizeMouseDown) // and it had been dragging a resize before.
+                {
+                    resizeMouseDown = false;
+                    // Resize by integer character cells, not by actual x/y pixels:
+                    // Note I wanted to call this dynamically as it's resizing, but there's some weird issue
+                    // with the timing of it, where the windowRect is temporarily set to a bogus value during the
+                    // time it was trying to run this, and thus it created exception-throwing code unless it waits
+                    // until the drag is done before trying to calculate this.
+                    // The effect the user sees is that the text can float in space past the edge of the window (when
+                    // shrinking the size) until the resize drag is done, and then it redraws itself.
+                    shared.Screen.SetSize(HowManyRowsFit(), HowManyColumnsFit());
+                }
+            }
+        }
```

Comment:
This is quite similar, but not *identical* logic to the resize feature of the source code editor popup window.  It's similar enough that we might consider trying to consolidate the logic together at some point.  It's one of those cases of almost-but-not-quite cut&paste code that makes you think "this is probably an opportunity for a refactor".